### PR TITLE
storageclassrequest: replace the filter logic with fieldIndexer

### PR DESCRIPTION
**Pull request by rewantsoni**
Use fieldIndexer for efficient querying of owned svgs instead of using a custom filter function

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -159,6 +159,22 @@ func main() {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
+	if err := mgr.GetCache().IndexField(
+		context.TODO(),
+		&cephv1.CephFilesystemSubVolumeGroup{},
+		util.CephFilesystemSubVolumeGroupOwnerKey,
+		func(obj apiclient.Object) []string {
+			refs := obj.GetOwnerReferences()
+			var owners []string
```
Comment (nb-ohad) [inline main.go:168]:
```suggestion
			owners := []string{}
```

File: main.go [hunk 2]
```diff
@@ -159,6 +159,22 @@ func main() {
...
+			for i := range refs {
+				owners = append(owners, string(refs[i].UID))
+			}
+			return owners
+		},
+	); err != nil {
+		setupLog.Error(err, "unable to setup FieldIndexer")
```
Comment (nb-ohad) [inline main.go:175]:
```suggestion
		setupLog.Error(err, "unable to set up FieldIndexer for owner reference UIDs")
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

**Review by leelavg — COMMENTED** [review]
I couldn't unsee Ohad's review and I don't have new comments to add.

File: controllers/storageclassrequest/storageclassrequest_controller.go [hunk 3]
```diff
@@ -46,6 +45,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+const ownerUID = "ownerUID"
```
Comment (nb-ohad) [inline controllers/storageclassrequest/storageclassrequest_controller.go:48]:
```suggestion
const ownerUID = "ownerUIDIndexName"
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: controllers/storageclassrequest/storageclassrequest_controller.go (hunk 3 above)
Comment (rewantsoni) [reply controllers/storageclassrequest/storageclassrequest_controller.go]: updated

**Pull request merged**